### PR TITLE
refactor(js-api): remove the JSON-RPC backend and unimplemented methods

### PR DESCRIPTION
**Pull request by leops**
## Summary

This PR aims at improving the usability of the `@rometools/js-api` package by cleaning up some unfinished features.

I removed the JSON-RPC backend as it's unlikely to be useful in the short term while we don't have a long running development server (since this only leaves the WASM backend, the concept of backend has instead been replaced with "distributions" to let the user choose which of the `bundler`, `nodejs` or `web` versions of `@rometools/wasm` they want to load).

I also removed the `formatFiles` and `parseContent` methods since those were not implemented at the moment. On the other hand I added new `lintContent` and `printDiagnostics` methods since those were mostly straightforward to implement and could provide additional value to the JS API. Finally I've added a `shutdown` method to manually free the WASM Workspace instance, as well as an experimental use of the `FinalizationRegistry` to automatically free some Workspace instances that are no longer reachable.

## Test Plan

I've updated the existing tests for the JS API to remove the daemon-specific tests and cover the new methods.

### CODE REVIEW

File: npm/js-api/package.json [hunk 1]
```diff
@@ -40,24 +40,27 @@
...
     "vite": "^3.0.8",
+    "@rometools/wasm-bundler": "../wasm-bundler",
     "@rometools/wasm-nodejs": "../wasm-nodejs",
-    "@rometools/backend-jsonrpc": "../backend-jsonrpc",
-    "rome": "../rome"
+    "@rometools/wasm-web": "../wasm-web"
   },
   "peerDependencies": {
-    "rome": "^10.0.0",
+    "@rometools/wasm-bundler": "^10.0.0",
```
Comment (ematipico) [inline npm/js-api/package.json:48]: Any reason why `rome` was removed from here?

File: npm/js-api/src/wasm.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	 * Use this if you want to communicate with the WebAssembly client built for the Web
+	 */
+	WEB,
+}
+
+let isInitialized = {
+	[Distribution.BUNDLER]: false,
+	[Distribution.NODE]: false,
+	[Distribution.WEB]: false,
+};
```
Comment (ematipico) [inline npm/js-api/src/wasm.ts:29]: This shape makes me think that more than one distribution can be initialized at the same time. Is it the case?

File: npm/js-api/tsconfig.json [hunk 3]
```diff
@@ -1,6 +1,7 @@
 {
   "compilerOptions": {
     "target": "es2020",                                  /* Set the JavaScript language version for emitted JavaScript and include compatible library declarations. */
+    "lib": ["ES2020", "ESNext.WeakRef"],
```
Comment (ematipico) [inline npm/js-api/tsconfig.json:4]: Is `WeakRef` used to enable the `FinalizationRegistry` feature?

**Review by ematipico — APPROVED** [review]

File: npm/js-api/package.json (hunk 1 above)
Comment (leops) [reply npm/js-api/package.json:48]: The `rome` package isn't being imported in the JS API, it only serves to pull in the pre-built binaries but we're not making use of that anywhere

File: npm/js-api/src/wasm.ts (hunk 2 above)
Comment (leops) [reply npm/js-api/src/wasm.ts:29]: Yes potentially someone could call `Rome.create` multiple times with different distribution, it's unclear why anyone would do that but I figured the modules should at least be initialized properly in that case ...

File: npm/js-api/tsconfig.json (hunk 3 above)
Comment (leops) [reply npm/js-api/tsconfig.json:4]: Yes `FinalizationRegistry` isn't part of the ES2020 specification

File: npm/js-api/package.json (hunk 1 above)
Comment (ematipico) [reply npm/js-api/package.json:48]: Then it should be removed from the `peerDependenciesMeta`

**Review by MichaReiser — APPROVED** [review]
Looks good. Please update the README with the correct installation and usage instructions.

**Pull request merged**